account_invoice_partner_number/models/account_invoice.py

```diff
@@ -1,11 +1,56 @@
 # -*- coding: utf-8 -*-
 # Copyright 2022 Quartile Limited
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+import logging
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class AccountInvoice(models.Model):
     _inherit = "account.invoice"
 
-    partner_no = fields.Char(related="partner_id.partner_no", readonly=True, store=True)
+    partner_no = fields.Char(string='Partner Number', readonly=True, store=True, compute='_compute_partner_number')
+
+    @api.depends('partner_id')
+    def _compute_partner_number(self):
+        logging.info("Call Compute Function")
+        for rec in self:
+            if rec.partner_id and rec.partner_id.partner_no:
+                if len(rec.partner_id.partner_no) < 6:
+                    length = 6 - len(rec.partner_id.partner_no)
+                    p_no = rec.partner_id.partner_no
+                    for i in range(0, length):
+                        p_no = '0' + p_no
+                    rec.partner_no = p_no
+                else:
+                    rec.partner_no = rec.partner_id.partner_no
+            else:
+                rec.partner_no = ""
```

Comment:
This is overly complicated.  Something like below should do (untested).

```suggestion
    @api.depends("commercial_partner_id", "commercial_partner_id.partner_no")
    def _compute_partner_no(self):
        for rec in self:
            if rec.type not in ("in_invoice", "in_refund", "in_receipt"):
                continue
            partner = rec.commercial_partner_id
            partner_no = "000000"
            if partner and partner.partner_no:
                partner_no += partner.partner_no
            rec.partner_no = partner_no[-6:]
```

@kanda999 I think it's more appropriate to use `commercial_partner_id` instead of `partner_id`.  Please confirm this point.

Reply:
zfill() looks simpler as @kanda999 suggested in Slack.